priv/worker.py

```diff
@@ -140,7 +156,7 @@ def env_close(self, instance_id):
       self._remove_env(instance_id)
 
   def env_close_all(self):
-    for key in self.envs.keys():
+    for key in list(self.envs.keys()):
```

Comment:
It's not a good idea to remove elements from the list while iterating over it, since you basically use an iterator. But I guess we could use a while loop what do you think?